### DISCUSSION TITLE
Mimir S3 storage format

### DISCUSSION THREAD

**Discussion by jochbru** [grafana/mimir#3668] [Help and support] [answered]
Hey everyone,

I'm using Grafana Mimir and have been impressed by it's performance and ease of use.

I'm currently configuring  the Mimir ruler with a simple configmap but this is reaching it's maximum configmap size of 1MB. When using this S3 storage backend for the ruler, in which format are the rules stored? Is this the regular yaml format and would they allow for direct editing in the bucket or does provisioning have to be performed through the ruler API?

**Comment by pstibrany** [comment, marked as answer]
Hi, rules stored in the bucket are not in YAML format. Instead they are [protobuf messages](https://github.com/grafana/mimir/blob/0d0070c721661f8928b86fc88ba9d80df6f6380b/pkg/ruler/rulespb/rules.proto#L21), and YAML files uploaded via API are converted to this format. This makes them difficult to edit directly.

**Reply by pracucci** [reply]
> or does provisioning have to be performed through the ruler API?

We also provide a CLI tool called `mimirtool` which allows you to upload the ruler YAML config to Mimir, and then Mimir will store it in the configured storage (typically the object storage):
https://grafana.com/docs/mimir/latest/operators-guide/tools/mimirtool/#rules

**Reply by jochbru** [reply]
Thanks for the suggestion @pracucci. I ended up using mimirtool with the sync function from a gitlab ci pipeline to manage the rules.
